### PR TITLE
Add an AGENTS.md

### PR DESCRIPTION
**Pull request by kronn**
Since AI-tooling is everywhere, I want to at least, give some guidance and support.

I would welcome feedback from different setups. Especially: are the instructions working?

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Overview
+
+This project is a web-application with the following stack:
+
+- Ruby
+- Ruby On Rails
+- PostgreSQL
+- Delayed::Job
```
Comment (carlobeltrame) [inline AGENTS.md:8]: Add Redis?

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+The Testing Framework is rspec, with capybara.
+Static analysis ist done with rubocop and brakeman.
+
+# Architecture
+
+hitobito consists of a core and one or more wagons. Wagons are mostly Rails' engines, just with a different loading order. See the [wagon integration doc](./doc/architecture/wagons/README.md) if information is needed. In the dev-setup, the needed wagons are located next to the core, so the core (this repo) is at `../hitobito/` while the wagons are in `../hitobito_*`. In some setups, only the currently needed ones are there, in others, all wagons are there, and the needed ones are selected by the ENV-Var `WAGONS`. In ruby, this is defined in the `Wagonfile`, which is included into the `Gemfile`.
+
+When asked to research anything, also consider the currently used wagons.
+
+The wagon always contains the group-structure of the final application and all modifications needed by that final application. The final application is also called the instance or the client-application (because our customer/client wants and needs that application).
```
Comment (carlobeltrame) [inline AGENTS.md:36]: Before talking about groups, should we mention what hitobito is used for? The term groups could refer to a lot of different things otherwise.

File: doc/developer/workflows/debugging.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+## Debugging
```
Comment (carlobeltrame) [inline doc/developer/workflows/debugging.md]: The instructions in here could more accurately be named "Bug fixing" imo. Debugging as I understand it is anything that involves pry or other brekapoints.

File: .agents/workflows/development.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+## Development
+
+The development workflow is the default. It is meant for new and changed functionality.
```
Comment (carlobeltrame) [inline .agents/workflows/development.md]: Once this first sentence is read, the LLM has already opted for this workflow. How baout you instead label `(default)` in the AGENTS.md?

File: doc/developer/workflows/development.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+The development workflow is the default. It is meant for new and changed functionality.
+
+- Start with a reason for the change, a goal that defines how the application should be improved for the user
+- Always write a spec first
+- The spec should define the desired state
+- The spec should be failing at first
+- Then add the implementation
+- The spec should now be successful
+- The whole spec-suite should also still be without errors
```
Comment (carlobeltrame) [inline doc/developer/workflows/development.md]: Do we really want the LLM to run the whole test suite on its own?

File: doc/developer/workflows/development.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+- You can run `rspec` to ensure that
```
Comment (carlobeltrame) [inline doc/developer/workflows/development.md]: These commands are not as easy to run in the docker dev setup. Maybe we need a parallel AGENTS.md there anyways..?

File: doc/developer/workflows/refactoring.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+## Refactoring
+
+This workflow should not change the functionality, but improve the internal structure. The goal could be a future extension, performance gains or better maintenance.
+
+As such, the addition of specs is optional. Without new specs, nothing may fail at first. For added abstractions, they new specs still allowed and needed.
```
Comment (carlobeltrame) [inline doc/developer/workflows/refactoring.md]: Last sentence is nonsensical

File: doc/developer/workflows/refactoring.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+Otherwise, the development workflow above still applies.
```
Comment (carlobeltrame) [inline doc/developer/workflows/refactoring.md]: The development workflow isn't above, it's in another file

**Comment by codez** [comment]
Könnten die Workflows nicht in einem `.agents` Verzeichnis abgelegt werden? Ich würde da schon einen Unterschied machen, ob das ein Dev lesen und befolgen soll oder die KI. Allenfalls können dann auch Symlinks für die populärsten Tools gemacht werden, falls sie das sonst nicht finden.

File: doc/developer/workflows/development.md (hunk 6 above)
Comment (kronn) [reply doc/developer/workflows/development.md]: Maybe there could be an additional AGENTS.md in the docker-setup that could be the primary one if using that. This can then include the one in the core-directory.

File: AGENTS.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+- Ruby On Rails
+- PostgreSQL
+- Delayed::Job
+- Redis
+
+A brief overview of the application is in the README.md.
+
+The HTML is mostly generated server-side with HAML.
+The Testing Framework is rspec, with capybara.
+Static analysis ist done with rubocop and brakeman.
```
Comment (njaeggi) [inline AGENTS.md]:
```suggestion
Static analysis is done with rubocop and brakeman.
```

File: AGENTS.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+# Overview
+
+This project is a web-application with the following stack:
+
+- Ruby
+- Ruby On Rails
```
Comment (njaeggi) [inline AGENTS.md:6]: Ich denke es könnte noch hilfreich sein hier zu erwähnen das `Webpacker` für das Asset Bundling gebraucht wird.

File: .agents/workflows/development.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+- Start with a reason for the change, a goal that defines how the application should be improved for the user
+- Always write a spec first
+- The spec should define the desired state
+- The spec should be failing at first
+- Then add the implementation
+- The spec should now be successful
+- Run the specs for all touched classes to avoid regressions
+- Then, run `brakeman` to ensure no new security-problem have been added
+- Lastly, run `rubocop` to ensure the code-style is good
+- Create a commit-message, summarizing the need for the change
```
Comment (njaeggi) [inline .agents/workflows/development.md:16]: Wollen wir hier noch erwähnen das wir nur Deutsch verwalten in den Translations, Agents fügen sonst sehr gerne bei allen Sprachen Translations ein 🙃

**Review by njaeggi — COMMENTED** [review]
Alles andere wurde bereits erwähnt/korrigiert, wie bereits besprochen brauchen wir im Docker Setup dann noch ein eigenes AGENTS.md 😄

File: .agents/workflows/development.md (hunk 11 above)
Comment (kronn) [reply .agents/workflows/development.md:16]: nice one. Ich habe das für den Setup in der AGENTS.md und nochmal in den Workflows erwähnt.

File: .agents/workflows/refactoring.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+## Refactoring
+
+This workflow should not change the functionality, but improve the internal structure. The goal could be a future extension, performance gains or better maintenance.
+
+Clarify the goal.
+
+Generally, the [development workflow](./development.md) applies. Read that file. New specs are optional with this workflow.
```
Comment (daniel-illi) [inline .agents/workflows/refactoring.md]: @kronn vielleicht könnten wir den agent noch instruieren zu prüfen, ob noch tests für offensichtliche edge cases fehlen

**Review by daniel-illi — APPROVED** [review]
LGTM 👍 

IMO sind der `development` workflow und der `fixing-a-bug` workflow sehr ähnlich, haben wir da einen Mehrwert beide zu pflegen?

File: .agents/workflows/refactoring.md (hunk 12 above)
Comment (kronn) [reply .agents/workflows/refactoring.md]: könnte zum token-burner werden. Ich füge es mal hinzu, notfalls, ändern wir es auf "weise auf offensichtlich fehlende Edge-cases hin".

**Comment by kronn** [comment]
> IMO sind der `development` workflow und der `fixing-a-bug` workflow sehr ähnlich, haben wir da einen Mehrwert beide zu pflegen?

Aktuell sind sie recht ähnlich. Ich vermute aber, dass wir verschiedene Schwerpunkte setzen wollen. Aus menschlicher Sicht ist der Unterschied nicht so groß, aber wir haben ja auch einen deutlich größeren impliziten Kontext und die Selbsreflexion, bei Problemen direkt den richtigen zu fragen. Ich vermute, dass wir gerade das Bug-Fixing noch verbessern können, wenn wir hier mehr Instruktionen reinwerfen, aber das sollte die Zeit zeigen.

Ich sehe das auch nicht als letzte Version, sondern recht bewusst als grobe erste Version. In der Praxis wird sich dann zeigen, wie man dieses project-prompt noch verbessern kann.

Mit etwas Glück löschen wir es einfach in einem Jahr wieder und alle sind damit happy. :joy:
